miniapps/nurbs/nurbs_ex1.cpp

```diff
@@ -10,6 +10,9 @@
 //               nurbs_ex1 -m ../../data/disc-nurbs.mesh -o -1
 //               nurbs_ex1 -m ../../data/pipe-nurbs.mesh -o -1
 //               nurbs_ex1 -m ../../data/beam-hex-nurbs.mesh -pm 1 -ps 2
+//               nurbs_ex1 -m ../../data/two-squares-nurbs.mesh -o 1 -rf ../../data/two-squares.ref
+//               nurbs_ex1 -m ../../data/two-squares-nurbs-rot.mesh -o 1 -rf ../../data/two-squares.ref
```

Comment:
For `two-squares-nurbs-rot.mesh`, the result cannot be visualized with glvis. Also, there is no error computation, so how can the solution be verified?

Reply:
I admit that this is problematic and I will think about this. I have compiled VisIT with the current branch which I can use to visualize the solution. In a visual comparison the solution is identical. 
I understand that compiling VisIt is too much to ask for a review.

Reply:
Because mesh (and solution?) files are slightly different this is an issue with visualizers that use the mfem package.
These need to be recompiled. For visit this can be a bit painful, glvis should be fine (i think and hope.)
@dylan-copeland did you recompile glvis with the appropriate mfem?

Reply:
Thanks @IdoAkkerman for the suggestion. I wasn't sure whether that should work, or whether glvis may need some code changes. Recompiling glvis did work for all the new meshes in this PR.

Reply:
@JacobLotz thanks for adding the new 3D sample runs. After rebuilding, glvis can plot the 2D mesh and solution, but in 3D it cannot plot the mesh with the solution. The 3D `refined.mesh` can be visualized with glvis, but not `sol.gf`.

Reply:
I have just compiled the 3D cases and I can actually plot the results for the 3D cases. However, I cannot find a slicing method in GlVis, so for the 3D cases I can only see the Dirichlet boundary...

I have created slices in VisIt orthogonal to the z-plane at z=0.5 for the results using the meshes `data/two-cubes-nurbs.mesh` and `data/two-cubes-nurbs-rot.mesh`. I applied 3 uniform refinements for the simulations, the automated refinement gave a too course mesh. (I will update the sample runs.) 

**`data/two-cubes-nurbs.mesh`**
![image](https://user-images.githubusercontent.com/69582468/217482717-d250580a-42ff-4205-aaee-cfa4f2533ec0.png)

**`data/two-cubes-nurbs-rot.mesh`**
![image](https://user-images.githubusercontent.com/69582468/217482890-624b9de9-43a7-4d7c-ab80-24adb2ae8fbb.png)

Reply:
Strangely, we seem to be getting different results. I rebuilt a fresh clone of glvis with this branch of MFEM, and the 2D solutions work but not the 3D. For the 3D tests, `glvis -m refined.mesh` works fine, but `glvis -m refined.mesh -g sol.gf` results in
NURBSFECollection: unknown geometry type.
Aborted (core dumped)

Anyway, the 2D results are good enough for now, and hopefully I will figure out my 3D issue at some point. I will resolve this thread.